tests/unit/api/github.test.ts

```diff
@@ -0,0 +1,104 @@
+import * as mongodb from 'mongodb';
+import { createSha256Signature } from '../../utils/createSha256Signature';
+import { MarkBuildArtifactsForDeletion } from '../../../api/controllers/v1/github';
+import { createMockAPIGatewayEvent } from '../../utils/createMockAPIGatewayEvent';
+
+const DELETION_SECRET = 'GH_DELETION_SECRET';
+
+// Mock RepoBranchesRepository so that we can mock which data to return.
+jest.mock('../../../src/repositories/repoBranchesRepository', () => ({
+  RepoBranchesRepository: jest.fn().mockImplementation(() => ({
+    getProjectByRepoName: jest.fn().mockImplementation((repoName) => repoName),
+  })),
+}));
+
+// Mock MetadataRepository so that we can mock which data to return.
+jest.mock('../../../src/repositories/metadataRepository', () => ({
+  MetadataRepository: jest.fn().mockImplementation(() => ({
+    marksMetadataForDeletion: jest.fn(),
+  })),
+}));
+
+// Mock UpdatedDocsRepository so that we can mock which data to return.
+jest.mock('../../../src/repositories/updatedDocsRepository', () => ({
+  UpdatedDocsRepository: jest.fn().mockImplementation(() => ({
+    marksAstsForDeletion: jest.fn(),
+  })),
+}));
+
+jest.mock('config', () => ({
+  get: () => DELETION_SECRET,
+}));
+
+describe('GitHub API Tests', () => {
+  describe('Artifact deletions', () => {
+    // Subset of data needed by the webhook. GH provides a lot more info.
+    const successfulPayload = {
+      action: 'closed',
+      pull_request: {
+        head: {
+          ref: 'test-webhook',
+        },
+      },
+      repository: {
+        name: 'docs-landing',
+      },
+    };
+
+    beforeAll(() => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      jest.spyOn(mongodb, 'MongoClient').mockReturnValue({
```

Comment:
I think ideally, we use `MongoMemoryServer` to have an actual local db to test against, but I think this can be done as part of a future isolated change. It would be beneficial for all of the other tests that currently mock some sort of repository